### PR TITLE
docs: add `defaultSFCLocale` and `globalSFCScope` to sfc

### LINKED ISSUE #554: add docs for 2 new options  from vite-plugin-vue-i18n
from issue comment

-----
I need to add docs for these 2 new options (but first I'll wait for your feedback, once ok I can add docs) and also on the site (at least here: https://vue-i18n.intlify.dev/guide/advanced/sfc.html#bundling-with-vite, and here https://vue-i18n.intlify.dev/guide/advanced/sfc.html#bundling-with-rollup (for webpack I need to review options), but it is is another repo).

_Originally posted by @userquin in https://github.com/intlify/vite-plugin-vue-i18n/issues/109#issuecomment-871253865_

### PR DESCRIPTION
**Pull request by userquin**
Closes #554 

I haven't tested on local, and I need to copy entries to `docs/ja/guide/advanced/sfc.md`.

Can I add `vitepress` to `dev dependencies` (what version have you installed globally?). 

I'm not a fan of using global dependencies...

### CODE REVIEW

**Comment by userquin** [comment]
Upps, I need to review this PR since those 2 options only added to vite plugin.

**Comment by userquin** [comment]
Maybe I can add those 2 options to webpack and rollup plugins...

**Comment by userquin** [comment]
The original rollup plugin seems to be not working, there are tests that fail, we need to fix the plugin:

```shell
 FAIL  test/resource-compilation.test.ts
  ● json resource

    Unexpected token (Note that you need @rollup/plugin-json to import JSON files)

      at error (node_modules/rollup/dist/shared/rollup.js:5305:30)
      at Module.error (node_modules/rollup/dist/shared/rollup.js:9750:16)
      at Module.tryParse (node_modules/rollup/dist/shared/rollup.js:10156:25)
      at Module.setSource (node_modules/rollup/dist/shared/rollup.js:10057:24)
      at ModuleLoader.addModuleSource (node_modules/rollup/dist/shared/rollup.js:18403:20)
      at ModuleLoader.fetchModule (node_modules/rollup/dist/shared/rollup.js:18459:9)
          at async Promise.all (index 0)
      at ModuleLoader.fetchStaticDependencies (node_modules/rollup/dist/shared/rollup.js:18485:34)
          at async Promise.all (index 0)
      at ModuleLoader.fetchModule (node_modules/rollup/dist/shared/rollup.js:18461:9)

  ● json5 resource

    'default' is not exported by test\fixtures\locales\en.json5, imported by test\fixtures\entry.js

      at error (node_modules/rollup/dist/shared/rollup.js:5305:30)
      at Module.error (node_modules/rollup/dist/shared/rollup.js:9750:16)
      at Module.traceVariable (node_modules/rollup/dist/shared/rollup.js:10138:29)
      at ModuleScope.findVariable (node_modules/rollup/dist/shared/rollup.js:8899:39)
      at TrackingScope.findVariable (node_modules/rollup/dist/shared/rollup.js:2647:38)
      at BlockScope.findVariable (node_modules/rollup/dist/shared/rollup.js:2647:38)
      at Identifier.bind (node_modules/rollup/dist/shared/rollup.js:4035:40)
      at AssignmentExpression.bind (node_modules/rollup/dist/shared/rollup.js:2734:23)
      at ExpressionStatement.bind (node_modules/rollup/dist/shared/rollup.js:2734:23)
      at BlockStatement.bind (node_modules/rollup/dist/shared/rollup.js:2730:31)

  ● yaml resource

    'default' is not exported by test\fixtures\locales\ko.yaml, imported by test\fixtures\entry.js

      at error (node_modules/rollup/dist/shared/rollup.js:5305:30)
      at Module.error (node_modules/rollup/dist/shared/rollup.js:9750:16)
      at Module.traceVariable (node_modules/rollup/dist/shared/rollup.js:10138:29)
      at ModuleScope.findVariable (node_modules/rollup/dist/shared/rollup.js:8899:39)
      at TrackingScope.findVariable (node_modules/rollup/dist/shared/rollup.js:2647:38)
      at BlockScope.findVariable (node_modules/rollup/dist/shared/rollup.js:2647:38)
      at Identifier.bind (node_modules/rollup/dist/shared/rollup.js:4035:40)
      at AssignmentExpression.bind (node_modules/rollup/dist/shared/rollup.js:2734:23)
      at ExpressionStatement.bind (node_modules/rollup/dist/shared/rollup.js:2734:23)
      at BlockStatement.bind (node_modules/rollup/dist/shared/rollup.js:2730:31)

  ● yml resource

    'default' is not exported by test\fixtures\locales\fr.yml, imported by test\fixtures\entry.js

      at error (node_modules/rollup/dist/shared/rollup.js:5305:30)
      at Module.error (node_modules/rollup/dist/shared/rollup.js:9750:16)
      at Module.traceVariable (node_modules/rollup/dist/shared/rollup.js:10138:29)
      at ModuleScope.findVariable (node_modules/rollup/dist/shared/rollup.js:8899:39)
      at TrackingScope.findVariable (node_modules/rollup/dist/shared/rollup.js:2647:38)
      at BlockScope.findVariable (node_modules/rollup/dist/shared/rollup.js:2647:38)
      at Identifier.bind (node_modules/rollup/dist/shared/rollup.js:4035:40)
      at AssignmentExpression.bind (node_modules/rollup/dist/shared/rollup.js:2734:23)
      at ExpressionStatement.bind (node_modules/rollup/dist/shared/rollup.js:2734:23)
      at BlockStatement.bind (node_modules/rollup/dist/shared/rollup.js:2730:31)

 PASS  test/custom-block.test.ts
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------|---------|----------|---------|---------|-------------------
All files |   91.84 |    81.82 |      75 |   91.67 |                  
 index.ts |   91.84 |    81.82 |      75 |   91.67 | 158-161,172,207  
----------|---------|----------|---------|---------|-------------------

Test Suites: 1 failed, 1 passed, 2 total
Tests:       4 failed, 10 passed, 14 total
```

**Comment by userquin** [comment]
@kazupon ping

File: docs/guide/advanced/sfc.md
Comment (kazupon) [inline docs/guide/advanced/sfc.md:306]: Unfortunately, we can use `globalSFCScope` and `defaultSFCLocale` options, so `@intlify/rollup-plugin-vue-i18n` only.
We need to mention at docs with `:::warning:::`

File: docs/guide/advanced/sfc.md
Comment (kazupon) [inline docs/guide/advanced/sfc.md:356]: In webpack, We don't still develop `defaultSSFCLang`.
Currently, we have a feature request.
https://github.com/intlify/bundle-tools/issues/24

File: docs/guide/advanced/sfc.md
Comment (kazupon) [inline docs/guide/advanced/sfc.md:406]: Same as the above my comment, we can use globalSFCScope and defaultSFCLocale options, so @intlify/rollup-plugin-vue-i18n only.
We need to mention at docs with :::warning:::

File: docs/guide/advanced/sfc.md
Comment (kazupon) [inline docs/guide/advanced/sfc.md:464]: Same as the above my comment, In webpack, We don't still develop `defaultSSFCLang`.
Currently, we have a feature request.
https://github.com/intlify/bundle-tools/issues/24

**Review by kazupon — CHANGES_REQUESTED** [review]
Thank you for your contribution!
I've just posted some comments.
Please check it! :)

**Comment by userquin** [comment]
Then, remove it from the docs or just add the warnings? Maybe it will be better to add `WIP` fro `rollup`  and `webpack` and so when done just remove those `WIP/Warning`.

**Review by kazupon — APPROVED** [review]

**Comment by kazupon** [comment]
@userquin 
Thank you for your updates!
I'll merge your PR once.
After I merge it, I'll tweaks docs.

Thanks!

**Pull request merged**
